### PR TITLE
Can specify more extra tokens that should be included in vocab in 'from_embeddings'

### PR DESCRIPTION
**Pull request by cle-ros**
- added the ability to provide arbitrary special tokens to the from_embeddings method of textfield

- added unit tests for this case

### CODE REVIEW

File: tests/unit/field/test_text_field.py [hunk 1]
```diff
@@ -242,6 +242,20 @@ def test_load_embeddings():
...
+        embeddings="tests/data/dummy_embeddings/test.txt",
+        pad_token=None,
+        unk_init_all=False,
+        additional_vocab_tokens=['<a>', '<b>', '<c>']
+    )
+    dummy = "a test ! <a> <b> "
+    field.setup([dummy])
+    assert '<a>' in field.vocab and '<b>' in field.vocab
+    assert '<c>' not in field.vocab
+    assert field.embedding_matrix[field.vocab['<a>']].size(-1) == 4
```
Comment (iitzco-asapp) [inline tests/unit/field/test_text_field.py]: assert than `<b>` is also part of the embedding_matrix?

File: tests/unit/field/test_text_field.py [hunk 2]
```diff
@@ -340,3 +354,31 @@ def test_text_process_unk():
     dummy = "justo Praesent luctus justo praesent"
     with pytest.raises(Exception):
         field.process(dummy)
+
+
+def recursive_tensor_to_list(data):
+    if isinstance(data, list):
+        return [recursive_tensor_to_list(d) for d in data]
+    elif isinstance(data, dict):
+        return dict((k, recursive_tensor_to_list(v)) for k, v in data.items())
```
Comment (iitzco-asapp) [inline tests/unit/field/test_text_field.py]:
```suggestion
        return {k: recursive_tensor_to_list(v)) for k, v in data.items()}
```

File: tests/unit/field/test_text_field.py [hunk 3]
```diff
@@ -340,3 +354,31 @@ def test_text_process_unk():
...
+        unk_init_all=False,
+        additional_vocab_tokens=['<a>', '<b>', '<c>']
+    )
+
+    dummy = "this is a test"
+    field.setup([dummy])
+    assert recursive_tensor_to_list(field.process(dummy)) == [1, 2, 3, 4]
+
+    dummy = "this is a test <a>"
+    assert recursive_tensor_to_list(field.process(dummy)) == [1, 2, 3, 4, 0]
```
Comment (iitzco-asapp) [inline tests/unit/field/test_text_field.py]: nit
For now this is fine, but its worth thinking how to decouple the implementation logic from the assertion (we are assuming that the `additional_vocab_tokens` are inserted in the vocab before the data tokens)

For example, we could do stuff like this
```
assert recursive_tensor_to_list(field.process(dummy))[-1] == field.vocab['<a>']
```

Comment (cle-ros) [reply tests/unit/field/test_text_field.py]: I agree. But that would be a general change to most of the different TextField tests.

File: tests/unit/field/test_text_field.py (hunk 1 above)
Comment (cle-ros) [reply tests/unit/field/test_text_field.py]: done

File: tests/unit/field/test_text_field.py (hunk 2 above)
Comment (cle-ros) [reply tests/unit/field/test_text_field.py]: done

**Comment by cle-ros** [comment]
replaced by https://github.com/asappresearch/flambe/pull/274

**Pull request closed without merging**